Logging in/register with Facebook...

justrau opened the pull request:
Logging in/registering with Facebook creates a user (if does not exist) or connects to an existing user if is logged in.

Note: src/DropTable/UserBundle/HWI/Bundle/OAuthBundle/OAuth/Response/PathUserResponse.php is not used at this point.

Commit:
Logging in/register with Facebook creates a user or connects with existing/logged-in one.

Commit:
Create UserService. Clean up templates and ProfileFromType a bit. Add AdminController with listAction to list all users at /admin/all-users.

Commit:
Rename UserService to UserManagerService.

Commit:
Add toggle enabled and locked in user management.

Commit:
Add switch roles to /admin/all-users.
Remove role entity.
Default last_login to '0' on /admin/all-users.

Commit:
Started creating template files. Login and register almost done, somewhy imported (as opposed to inline) <script> does not work, so Facebook popup dialog box does not work.
Commented out 127.0.0.1 requirement in app_dev.php for deployment into server.

Commit:
Disable writing of password request time to remove the constraint of password requesting once per 24 hours.

lukasvtechceo commented on src/DropTable/UserBundle/Controller/ResettingController.php:
```diff
@@ -0,0 +1,162 @@
+        $user = $this->get('fos_user.user_manager')->findUserByUsernameOrEmail($username);
+
+        if (null === $user) {
+            return $this->render('FOSUserBundle:Resetting:request.html.twig', array(
```
you can youse [] as array expresion

lukasvtechceo commented on src/DropTable/UserBundle/Controller/ResettingController.php:
```diff
@@ -0,0 +1,162 @@
+
+        $user = $userManager->findUserByConfirmationToken($token);
+
+        if (null === $user) {
```
You can abstract it as common method oi this controller or facade.
Why you cant use same code in sendEmailAction also?

justrau replied:
Since I am simply overriding FOSUserBundle controller, I only changed parts for the functionality I needed. Or should I abstract it as a common method anyway?

lukasvtechceo replied:
If you are having same or similar functionality its better to move it in one method and accesit it where you need it.
In this case I see same logic, but different exception/error/notification handling.
So it almost fits: http://code.tutsplus.com/tutorials/design-patterns-the-facade-pattern--cms-22238
But I'm not sure why exception is handled in different ways...

lukasvtechceo commented on src/DropTable/UserBundle/Controller/ResettingController.php:
```diff
@@ -0,0 +1,162 @@
+        $event = new GetResponseUserEvent($user, $request);
+        $dispatcher->dispatch(FOSUserEvents::RESETTING_RESET_INITIALIZE, $event);
+
+        if (null !== $event->getResponse()) {
```
I would write a comment about this line. Why execution is stopped here

lukasvtechceo commented on src/DropTable/UserBundle/Controller/ResettingController.php:
```diff
@@ -0,0 +1,162 @@
+            $userManager->updateUser($user);
+
+            if (null === $response = $event->getResponse()) {
+                $url = $this->generateUrl('fos_user_profile_show');
```
same here, why we are returning if there are some code left.
It's not mandatory to comment, bet in my oppinion it would make code clerear

lukasvtechceo commented on src/DropTable/UserBundle/Entity/User.php:
```diff
@@ -104,16 +105,24 @@ public function getLastName()
-        $this->createdAt = $createdAt;
-
-        return $this;
+        $this->createdAt = new \DateTime('now', new \DateTimeZone(('Europe/Vilnius')));
```
reconsider moving timezone to config or in worst case local timezone.

lukasvtechceo commented on src/DropTable/UserBundle/Resources/config/services.yml:
```diff
@@ -10,4 +13,20 @@ services:
+    user_manager:
+        class: %user_manager_class%
+        arguments:
+            - @doctrine.orm.entity_manager
```
new line add the on file is missing

lukasvtechceo commented on src/DropTable/UserBundle/Resources/public/css/custom.css:
```diff
@@ -0,0 +1,4 @@
+#login,
+#register {
+    margin-top: 18%;
+}
```
new line add the on file is missing

lukasvtechceo commented on src/DropTable/UserBundle/Resources/config/routing.yml:
```diff
@@ -0,0 +1,18 @@
+    defaults: { _controller: DropTableUserBundle:Admin:switchRole }
+
+home:
+    path: /
```
new line add the on file is missing

lukasvtechceo commented on src/DropTable/UserBundle/Resources/public/js/custom.js:
```diff
@@ -0,0 +1,38 @@
+            }, {scope: 'email'});
+        }
+    });
+}
```
new line add the on file is missing

lukasvtechceo commented on src/DropTable/UserBundle/Resources/views/Admin/all_users.html.twig:
```diff
@@ -0,0 +1,66 @@
+        </tr>
+    {% endfor %}
+    </table>
+{% endblock content %}
```
new line add the on file is missing

lukasvtechceo commented on src/DropTable/UserBundle/Resources/views/Default/admin_layout.html.twig:
```diff
@@ -0,0 +1,13 @@
+            </div>
+        </div>
+    </div>
+{% endblock body %}
```
new line add the on file is missing

lukasvtechceo commented on src/DropTable/UserBundle/Resources/views/Default/base.html.twig:
```diff
@@ -0,0 +1,36 @@
+            <script src="{{ asset_url }}"></script>
+        {% endjavascripts %}
+    </body>
+</html>
```
new line add the on file is missing

lukasvtechceo commented on src/DropTable/UserBundle/Resources/views/Default/fos_layout.html.twig:
```diff
@@ -0,0 +1,77 @@
+    {% render(controller('HWIOAuthBundle:Connect:connect')) %}
+
+    <a href="{{ path('fos_user_resetting_request') }}">Forgot password ?</a>
+{% endblock body %}
```
new line add the on file is missing

lukasvtechceo commented on src/DropTable/UserBundle/Resources/views/Default/layout.html.twig:
```diff
@@ -0,0 +1,15 @@
+            </div>
+        </div>
+    </div>
+{% endblock body %}
```
new line add the on file is missing

lukasvtechceo commented on src/DropTable/UserBundle/Security/Core/User/FOSUBUserProvider.php:
```diff
@@ -0,0 +1,85 @@
+        $email = $response->getEmail();
+        $user = $this->userManager->findUserBy([$this->getProperty($response) => $username]);
+        // When the user is registering.
+        if (null === $user) {
```
you can move this block to separate method

lukasvtechceo commented on src/DropTable/UserBundle/Security/Core/User/FOSUBUserProvider.php:
```diff
@@ -0,0 +1,85 @@
+        return $user;
+    }
+
+}
```
new line add the on file is missing

Commit:
Create createNewUser() in FOSUBUserProvider.
Create handleException() in ResettingController.
Fix Facebook login/register button.
Add some comments.
Add lines at EOF.

Commit:
Refactor handleException() ir resettingController.
